### ISSUE TITLE
Error Cannot find "vendor/bin/phpcs" on Windows

### ISSUE THREAD

**Issue by JustBlackBird** [JustBlackBird/gulp-phpcs#21]
Each of these `bin` option values

- `./vendor/bin/phpcs`
- `.\\vendor\\bin\\phpcs`
- `vendor/bin/phpcs`

end with `Error: Cannot find` error on Windows.

___

Directory structure:
```
gulpfile.js
...
\vendor
    \bin
        phpcs
```

Gulp code:
```
gulp.task('php_cs', function() {
    return gulp.src(['src/**/*.php'])
        // Validate files using PHP Code Sniffer
        .pipe(phpcs({
            bin: 'vendor/bin/phpcs',
            standard: 'PSR2',
            warningSeverity: 0
        }))
        // Log all problems that was found
        .pipe(phpcs.reporter('log'));
});
```

Results:
```
[13:37:16] Using gulpfile C:\xampp\htdocs\website\gulpfile.js
[13:37:16] Starting 'php_cs'...

events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: Cannot find "vendor/bin/phpcs"
    at ChildProcess.<anonymous> (C:\xampp\htdocs\website\node_modules\gulp-phpcs\index.js:81:32)
    at emitOne (events.js:77:13)
    at ChildProcess.emit (events.js:169:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
    at onErrorNT (internal/child_process.js:344:16)
    at doNTCallback2 (node.js:429:9)
    at process._tickDomainCallback (node.js:384:17)
```

Using gulp-phpcs 1.0.0

**Comment by JustBlackBird** [comment]
The problem is in files extensions. As I see, composer creates `vendor\bin\phpcs.bat` on windows instead of just `vendor\bin\phpcs`. Normally windows cmd resolves `command.bat` to `command`. At the same time, node.js `child_process.spawn` does not use shell to spawn PHPCS process, so no resolving is done.

As a temporary workaround you can specify `.\\vendor\\bin\\phpcs.bat` as `bin` option.

**Comment by tyler36** [comment]
This is the same issue I was having with 'PHPcs encoding issue #10'

For some reason, the error was outputting with incorrect coding, and therefore unreadable. I was assuming it was an encoding error with this plugin. Since PHPCS worked fine from CLI in windows and CLI / gulp via Homestead.

**Comment by JustBlackBird** [comment]
Well, the problem described in the issue is not only in slash styles. It's most about resolving command path _without_ running subshell (subshell is spawned by node.js `child_process.exec` but not with `child_process.spawn`).

**Comment by cedon** [comment]
I'm having this issue as well. My installation of phpcs is on my C: while my development stuff is on D:. My php directory is the path so setting `bin: 'phpcs.bat'` worked for me to fix it. It's something on Microsoft's end which will hopefully be fixed with Windows 10 getting BASH soon.

**Comment by JustBlackBird** [comment]
I'm not sure what your problem exactly is but setting full path as `bin` option should help.

**Comment by cedon** [comment]
It's the same issue of `phpcs.bat` not being resolved as `phpcs` when gulp. In a normal command prompt, it works fine with `phpcs {filename}`, but with gulp I get "Cannot find phpcs".

**Comment by JustBlackBird** [comment]
At the moment i'm waiting for https://github.com/npm/node-which/issues/29 and https://github.com/npm/node-which/issues/26 to be resolved to rewrite gulp-phpcs bin finder logic.

**Comment by JustBlackBird** [comment]
Fixed in [v1.1.0](https://github.com/JustBlackBird/gulp-phpcs/releases/tag/v1.1.0)
